js/src/ui/Form/TypedInput/typedInput.js

```diff
@@ -226,21 +237,23 @@ export default class TypedInput extends Component {
   }
 
   renderInteger (value = this.props.value, onChange = this.onChange) {
-    const { label, error, hint, min, max } = this.props;
+    const { allowCopy, label, error, hint, min, max, readOnly } = this.props;
     const param = this.getParam();
 
-    const realValue = value && typeof value.toNumber === 'function'
-      ? value.toNumber()
+    const realValue = value
+      ? (new BigNumber(value))[readOnly ? 'toFormat' : 'toNumber']()
       : value;
 
     return (
       <Input
+        allowCopy={ allowCopy }
         label={ label }
         hint={ hint }
         value={ realValue }
         error={ error }
         onChange={ onChange }
-        type='number'
+        readOnly={ readOnly }
+        type={ readOnly ? 'text' : 'number' }
```

Comment:
Because if the TypedInput is read only and a number, we display the formatted number (which must be in a text input then)